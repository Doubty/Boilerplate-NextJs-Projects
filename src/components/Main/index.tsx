import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="Logo do projeto - Boilerplate" />
    <S.Title>BoilerPlate do Galvs</S.Title>
    <S.Description>
      Boilerplate criado com amor e dedicação total de Antônio Galvão.
      Boilerplate conta com projeto em Next.js usando typescript e styled
      componentes, bem como, configurado para testes, prettier, eslint e husky.
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Desenvolvedor na frente do computador"
    />
  </S.Wrapper>
)

export default Main
