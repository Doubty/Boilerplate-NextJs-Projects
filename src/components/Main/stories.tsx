import Main from '.'
import { Meta, Story } from '@storybook/react'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Default title',
    description: 'Default description'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: 'Basic title',
  description: 'Basic description'
}

export const Default: Story = (args) => <Main {...args} />
