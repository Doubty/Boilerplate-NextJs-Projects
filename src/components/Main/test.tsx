import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('it shoudl render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /BoilerPlate do Galvs/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('it should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
