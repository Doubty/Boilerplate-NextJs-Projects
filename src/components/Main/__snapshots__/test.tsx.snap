// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> it shoudl render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2.5rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Logo do projeto - Boilerplate"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    BoilerPlate do Galvs
  </h1>
  <h2
    class="c3"
  >
    Boilerplate criado com amor e dedicação total de Antônio Galvão. Boilerplate conta com projeto em Next.js usando typescript e styled componentes, bem como, configurado para testes, prettier, eslint e husky.
  </h2>
  <img
    alt="Desenvolvedor na frente do computador"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
